src/bin/cairo_threads.rs

```diff
@@ -215,20 +216,17 @@ fn build_ui(application: &gtk::Application) {
         Inhibit(false)
     }));
 
-    // Check every 100ms if there are new images from any worker threads and if so queue the
-    // corresponding part of the DrawingArea for redrawing
-    gtk::timeout_add(100, move || {
-        while let Ok((thread_num, image)) = ready_rx.try_recv() {
-            let (ref images, ref origins, ref workers) = &*workspace;
+    ready_rx.attach(None, move |(thread_num, image)| {
+        let (ref images, ref origins, ref workers) = &*workspace;
```

Comment:
& seems derefed automatically and unneeded

Reply:
```
220 |         let (ref images, ref origins, ref workers) = workspace;
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `std::rc::Rc`, found tuple
    |
    = note: expected type `std::rc::Rc<(std::vec::Vec<std::cell::RefCell<Image>>, std::vec::Vec<(i32, i32)>, std::vec::Vec<std::sync::mpsc::Sender<Image>>)>`
               found type `(_, _, _)`
```
Not here? :)

Reply:
Oh you mean & is not needed. Indeed.